playbooks/openshift/heat_deprovision.yml

```diff
@@ -12,14 +12,15 @@
         server: "{{ cluster_name }}-lb-0"
         floating_ip_address: "{{ openshift_public_ip }}"
         state: absent
-      when: lb_vm_group_size > 0
+      when: master_vm_group_size > 1
     - name: Disassociate floating IP from first master node
       os_floating_ip:
         server: "{{ cluster_name }}-master-0"
         floating_ip_address: "{{ openshift_public_ip }}"
         state: absent
-      when: lb_vm_group_size == 0
+      when: master_vm_group_size == 1
```

Comment:
As I mentioned in the previous comment, this could cause issues when you have multiple masters that act as load balancers.